### DISCUSSION TITLE
Does OPA support JSON patch move operation?

### DISCUSSION THREAD

**Discussion by thilak009** [open-policy-agent/community#462] [OPA and Rego] [answered]
OPA does not seem to support the `move` operation of the JSON patch specification, the [docs](https://www.openpolicyagent.org/docs/latest/rest-api/#patch-a-document) mention the patch operation is based on RFC 6092 but `move` operation returns error response
```
{
    "code": "invalid_parameter",
    "message": "bad patch operation: move"
}
```
I see that only `add`,`remove` and `replace` [are supported ](https://github.com/open-policy-agent/opa/blob/main/server/server.go#L2560), any reason for this?
This [example](https://play.openpolicyagent.org/p/QqpxumAD6B) in playground seems to works as expected

**Comment by charlieegan3** [comment, marked as answer]
Hi @thilak009, you're right in that there is an inconsistency here.

> This [example](https://play.openpolicyagent.org/p/QqpxumAD6B) in playground seems to works as expected

The built in function implementation is different from the code which handles API requests for the Data patch operation.

> any reason for this?

The superficial reason is that these are all that's exposed by the underlying data storage interface, of which there are two in-tree implementations (inmem and disk). Changes to this interface would need to be carefully considered since there are likely 3rd party implementations that would break if we added required support for more operations.

What are you trying to do? There might be a better way to achieve it. Have you considered Delta Bundles? https://www.openpolicyagent.org/docs/latest/management-bundles/#delta-bundles

**Reply by thilak009** [reply]
Hi @charlieegan3, thanks for the quick reply, I am trying to implement PATCH support in [OPAL](https://github.com/permitio/opal) (which is an administration layer on top of OPA)

As part of the implementation, i was trying out all JSON patch operations according to the RFC and discovered that OPA does not support `move`, hence wanted to clarify

As for my use case, seems like OPAL will not be able to support `move` as a PATCH operation until OPA does

**Reply by charlieegan3** [reply]
I see! Yes, that's correct. I think that adding support for `move` is unlikely, however feel free to open an issue to discuss it in more detail if you'd like to make a case for it with a strong use case.
